src/components/Purchase/CheckoutPage.js

```diff
@@ -21,34 +22,40 @@ const total = 140.0;
 
 const CheckoutPage = (/*{ album, quantity, subtotal, tax, total}*/) => {
   const [clientSecret, setClientSecret] = useState('');
-
+  const [orderData, setOrderData] = useState(null);
+  const location = useLocation();
   useEffect(() => {
-    // Create PaymentIntent as soon as the page loads
-    const fetchData = async () => {
-      try {
-        const response = await axios.post(
-          'http://localhost:8000/api/v1/orders',
-          {
-            orderItems: [{ album, quantity }],
-            subtotal,
-            tax,
-            total,
-          },
-          {
-            withCredentials: true,
-            headers: { 'Content-Type': 'application/json' },
-          }
-        );
+    // Extract order data from navigation.state
+    const { orderData } = location.state || {};
 
-        setClientSecret(response.data.clientSecret);
-      } catch (error) {
-        // Handle error here
-        console.error('Error fetching data:', error);
-      }
-    };
+    if (orderData) {
+      setOrderData(orderData);
+      // Create PaymentIntent as soon as the page loads
+      const fetchData = async () => {
+        try {
+          const response = await axios.post(
+            `${process.env.REACT_APP_API_BASE_PATH}/orders`,
+            {
+              orderItems: orderData.orderItems, // Use order data received from state
+              subtotal: orderData.subtotal,
+              tax: orderData.tax,
+              total: orderData.total,
+            },
+            {
+              withCredentials: true,
+              headers: { 'Content-Type': 'application/json' },
+            }
+          );
 
-    fetchData();
-  }, [subtotal, album, quantity, tax, total, setClientSecret]);
+          setClientSecret(response.data.clientSecret);
+        } catch (error) {
+          // Handle error here
+          console.error('Error fetching data:', error);
+        }
+      };
+      fetchData();
+    }
+  }, [setClientSecret]);
```

Comment:
This main external dependency of this useEffect is now `location.state`. We want it so that this useEffect hook only runs when that value changes/is updated:

```suggestion
  }, [location.state]);
```

Reply:
Updated